### ISSUE TITLE
CORS middleware returns 404 if there was Origin header in request

### ISSUE THREAD

**Issue by mxl** [http4s/http4s#933]
After upgrade from 0.14.11 to 0.15.* CORS middleware returns 404 if there are more than 1 mounted services with CORS middleware and Origin header is presented in request:

```scala
import java.net.InetSocketAddress

import org.http4s._
import org.http4s.dsl._
import org.http4s.server.ServerApp
import org.http4s.server.blaze.BlazeBuilder
import org.http4s.server.middleware.{CORS, CORSConfig}

import scala.concurrent.duration._

object Main extends ServerApp {

  val cors = (s: HttpService) =>
    CORS(
      s,
      CORSConfig(
        anyOrigin = true,
        anyMethod = true,
        maxAge = 1.day.toSeconds,
        allowCredentials = true
      )
    )

  override def server(args: List[String]) =
    BlazeBuilder
      .mountService(cors(HttpService {
        case GET -> Root / "test1" => Ok()
      }))
      .mountService(cors(HttpService {
        case GET -> Root / "test2" => Ok()
      }))
      .bindSocketAddress(new InetSocketAddress(8000))
      .start
}
```

```
curl -H "Origin:*" http://localhost:8000/test1 
```

returns `404 Not Found`.

**Comment by mxl** [comment]
See https://github.com/mxl/scala-issues/tree/http4s-cors-issue example project (checkout http4s-cors-issue branch).

**Comment by mxl** [comment]
This bug is a showstopper 😢 @rossabaker Could you estimate when you will have a time to fix it? I can try to nail it if you are busy now.

**Comment by reactormonk** [comment]
Does it work with `0.15.3`? https://github.com/http4s/http4s/pull/810

**Comment by rossabaker** [comment]
#810 is introduced in 0.16, but I'll bet it's like #803.  I am on family duty, but am hopeful I can fix it this evening.

**Comment by ChristopherDavenport** [comment]
Closed by #934

**Comment by rossabaker** [comment]
The fix for this is released as v0.15.5 and should be in Maven Central soon.
